test/unit/logic/inTransfer.js

```diff
@@ -403,62 +397,61 @@ describe('inTransfer', function () {
 			});
 		});
 
-		// TODO: #1242 Have to disabled due to issue https://github.com/LiskHQ/lisk/issues/1242
-		describe.skip('when shared.getGenesis succeeds', function () {
+		describe('when shared.getGenesis succeeds', function () {
 
 			beforeEach(function () {
-				sharedStub.getGenesis = sinon.stub.callsArg(1);
+				sharedStub.getGenesis = sinonSandbox.stub().callsArg(1);
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet', function () {
 				expect(accountsStub.mergeAccountAndGet.calledOnce).to.be.true;
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet with address = dapp.authorId', function () {
-				expect(accountsStub.mergeAccountAndGet.calledWith(sinon.match({address: validGetGensisResult.authorId}))).to.be.true;
+				expect(accountsStub.mergeAccountAndGet.calledWith(sinonSandbox.match({address: validGetGensisResult.authorId}))).to.be.true;
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet with balance = trs.amount', function () {
-				expect(accountsStub.mergeAccountAndGet.calledWith(sinon.match({balance: trs.amount}))).to.be.true;
+				expect(accountsStub.mergeAccountAndGet.calledWith(sinonSandbox.match({balance: trs.amount}))).to.be.true;
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet with u_balance = trs.amount', function () {
-				expect(accountsStub.mergeAccountAndGet.calledWith(sinon.match({u_balance: trs.amount}))).to.be.true;
+				expect(accountsStub.mergeAccountAndGet.calledWith(sinonSandbox.match({u_balance: trs.amount}))).to.be.true;
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet with blockId = block.id', function () {
-				expect(accountsStub.mergeAccountAndGet.calledWith(sinon.match({blockId: dummyBlock.id}))).to.be.true;
+				expect(accountsStub.mergeAccountAndGet.calledWith(sinonSandbox.match({blockId: dummyBlock.id}))).to.be.true;
 			});
 
 			it('should call modules.accounts.mergeAccountAndGet with round = slots.calcRound result', function () {
-				expect(accountsStub.mergeAccountAndGet.calledWith(sinon.match({round: slots.calcRound(dummyBlock.height)}))).to.be.true;
+				expect(accountsStub.mergeAccountAndGet.calledWith(sinonSandbox.match({round: slots.calcRound(dummyBlock.height)}))).to.be.true;
 			});
+		});
 
-			describe('when modules.accounts.mergeAccountAndGet fails', function () {
+		describe('when modules.accounts.mergeAccountAndGet fails', function () {
```

Comment:
Are these definitely supposed to be nested at this level?

Reply:
yes, the intransfer undo was nested incorrectly due to which the tests where failing.